stdlib/public/Cxx/std/String.swift

```diff
@@ -34,7 +34,24 @@ extension std.string {
   }
 
   @_alwaysEmitIntoClient
-  public init(_ string: UnsafePointer<CChar>?) {
+  public init(_ string: String?) {
+    if let s = string {
+      unsafe self = unsafe s.withCString(encodedAs: UTF8.self) { buffer in
+  #if os(Windows)
+        // Use the 2 parameter constructor.
```

Comment:
Do we still need this `#if` now that we support `enable_if` in the clang importer?

Reply:
I don't think the support for `enable_if` is advanced enough yet to compile this. This is also old code so if we think it needs to change, I would do that in a separate PR.